Fixed "Issue" #1

tycebrown opened the pull request:
1. Refactored the HomePage into a ListWidget.
2. Set body to a list view.

tycebrown committed:
Fixed "Issue" #1

1. Refactored the HomePage into a ListWidget.
2. replaced body with a list view.

tycebrown commented:
Note that it has not been styled.

Pertempto commented:
Just quickly skimming over the code, it looks pretty good! When I get the chance to sit down at my computer and take a deeper look I'll do a full review. That might be this weekend or early next week.

Pertempto commented on lib/list_widget.dart:
```diff
@@ -0,0 +1,51 @@
+  }
+
+  void _addNewItem() {
+    final itemToAdd = ListTile(
+      title: Text(
+        'New item',
+        style: itemTextStyle,
+      ),
+    );
```
```suggestion
    const itemTextStyle = Theme.of(context).textTheme.titleLarge
    final itemToAdd = ListTile(
      title: Text(
        'New item',
        style: itemTextStyle,
      ),
    );
```
Instead of making your own text style, I would recommend using one of the built-in ones from [here](https://api.flutter.dev/flutter/material/TextTheme-class.html). Using the built-in ones will lead to more consistency in your apps.

tycebrown replied:
Done!

Pertempto commented on lib/list_widget.dart:
```diff
@@ -0,0 +1,51 @@
+import 'package:flutter/material.dart';
+
+class ListWidget extends StatefulWidget {
```
Include some kind of header comment above the widget that explains its purpose.

tycebrown replied:
Done!

Pertempto commented on lib/list_widget.dart:
```diff
@@ -0,0 +1,51 @@
+class _ListWidgetState extends State<ListWidget> {
+  static const itemTextStyle = TextStyle(fontSize: 20);
+
+  final list = <Widget>[];
```
This is the data part of your app. It is strange to store it as a list of Widgets. What if the user wants to edit one of the items? We need to write our code in such a way that we can easily add more features in the future. Instead, I would recommend storing it as a list of strings.

Pertempto replied:
Using descriptive variables names is a very important part of coding. `list` is about as vague as you can get. `List` is a built-in class in Dart, so you don't want a variable called `list`. A future reader of the code could easily be confused by this name. What if you named it something like `listItems`?

tycebrown replied:
Done!

Pertempto commented on lib/list_widget.dart:
```diff
@@ -0,0 +1,51 @@
+    // not be modified once it is passed in.
+    //
+    return ListView(
+      children: list.toList(),
```
If `list` was a list of strings, you could do something like `children: list.map((item) => _buildItemView(item))`

tycebrown replied:
Done!

Pertempto commented on lib/list_widget.dart:
```diff
@@ -0,0 +1,51 @@
+    );
+  }
+
+  void _addNewItem() {
```
In this method, instead of adding widgets to `list`, add a string. You can convert the strings to widgets at render-time in `_buildListView`.

tycebrown replied:
Done!

Pertempto commented:
@tycebrown I have completed my first review of this pull request. The functionality works great, the code just needs some cleanup.  
* Please address each of the items I mentioned above.
    * Respond with "done" or something similar to each item when they have been completed
         * Please do *not* click the "Resolve conversation" button
         * I will click it after I have confirmed that you have applied the necessary fix
* When you have addressed all the items, please remove the "changes requested" label, and leave a comment asking me to look over this again.

tycebrown committed:
Updated text style

tycebrown committed:
Included header comment

tycebrown committed:
Refactored list -> listItems

Changed the name of the list and made the necessary changes so that listItems stores Strings instead of Widgets.

tycebrown committed:
Tidied up code and silenced vscode's complaints.

tycebrown commented:
Done and ready!

Pertempto commented on lib/list_widget.dart:
```diff
@@ -0,0 +1,57 @@
+    );
+  }
+
+  ListView _buildListView() {
+    return ListView(
+      children: listItems
+          .map(
+            (item) => _buildItemWidget(item),
+          )
+          .toList(),
+    );
+  }
+
+  Widget _buildItemWidget(String item) {
+    final itemTextStyle = Theme.of(context).textTheme.titleLarge;
+    return ListTile(
+      title: Text(
+        item,
+        style: itemTextStyle,
+      ),
+    );
+  }
```
My personal preference is to keep my code in as least amount of lines as a I can (within reason).   
For these two methods, I would remove the trailing commas and write them as follows:  
```suggestion
  ListView _buildListView() => ListView(
      children: listItems.map((item) => _buildItemWidget(item)).toList());

  Widget _buildItemWidget(String item) {
    final itemTextStyle = Theme.of(context).textTheme.titleLarge;
    return ListTile(title: Text(item, style: itemTextStyle));
  }
```
That's 7 lines of code, instead of 19. This makes it easier to see more code on one screen, without having to scroll. This is just a style preference. If you find it easier to read as it is, then don't change anything and just leave a comment.

Pertempto commented on lib/list_widget.dart:
```diff
@@ -0,0 +1,57 @@
+    );
+  }
+
+  void _addNewItem() {
+    final itemToAdd = 'New Item';
+
+    setState(() => listItems.add(itemToAdd));
+  }
+}
```
```suggestion
  void _addNewItem() {
    final itemToAdd = 'New Item';
    setState(() => listItems.add(itemToAdd));
  }
}
```  
The extra blank line isn't really adding anything.

Pertempto commented on lib/list_widget.dart:
```diff
@@ -0,0 +1,57 @@
+import 'package:flutter/material.dart';
+
+///ListWidget: 
+///  - a widget representing a page containing a single list in the 
+///  app, such as a to-do list, a shopping list, etc.
+
+class ListWidget extends StatefulWidget {
```
```suggestion
/// ListWidget: 
///  - a widget representing a page containing a single list in the 
///  app, such as a to-do list, a shopping list, etc.
class ListWidget extends StatefulWidget {
```
Don't leave blank line after the comment, be sure to leave a space between the `///` and your words.

Pertempto commented:
Three style-related things to address, and then we'll be done here!

tycebrown committed:
Fixed styling

tycebrown commented:
Committed & pushed your style changes.

Pertempto reviewed (approved):
Everything looks great!  
Good work!!
